metadata-models/src/main/pegasus/com/linkedin/retention/Retention.pdl

```diff
@@ -0,0 +1,11 @@
+namespace com.linkedin.retention
+
+/**
+ * Base class that encapsulates different retention policies.
+ * Only one of the fields should be set
+ */
+record Retention {
```

Comment:
the other option is to have a type and arguments:

type: enum Retentiontype: {
   indefinite,
   version,
   time
}
// set if type is version 
maxVersion: optional long

// set if type is time 
maxAgeInseconds

this structure just makes reading the object in code a bit more complex.. is there a reason you went with the approach of just setting a field and letting the reader check the presence of each?

Reply:
Hmn since there is no shared fields, I think this approach is much easier to read. Otherwise, each type needs to be mapped to the field that needs to be filled out in code instead of in pdl.